tests/test_adjoint_jacobian.py

```diff
@@ -700,23 +700,20 @@ def dev(self, request):
         return qml.device(device_name, wires=2, c_dtype=request.param)
 
     @pytest.mark.skipif(ld._new_API, reason="Old API required")
-    def test_finite_shots_warning(self):
-        """Tests that a warning is raised when computing the adjoint diff on a device with finite shots"""
+    def test_finite_shots_error(self):
+        """Tests that an error is raised when computing the adjoint diff on a device with finite shots"""
 
         dev = qml.device(device_name, wires=1, shots=1)
 
-        with pytest.warns(
-            UserWarning, match="Requested adjoint differentiation to be computed with finite shots."
+        with pytest.raises(
+            qml.QuantumFunctionError, match="does not support adjoint with requested circuit."
```

Comment:
Sorry, but I could not find this `raise` message in the code. Could you double-check the match of this `raise` message, please?

Reply:
This error is rained in PennyLane, not in Lightning. That's why you will not find it in the code. 

As for the double-checking, `pytest` fails a test if an expected error is not raised, so you don't need to worry about that 😄

Reply:
The error is raised [here](https://github.com/PennyLaneAI/pennylane/blob/ad/facade-wrapper/pennylane/workflow/qnode.py#L706)

Reply:
Thank you for your comments, I thought if the raise message did not completely match with `pytest.raises()` `match` option, the test itself would fail.

Reply:
Yes it will. I am a bit confused about your concern now 😅

Reply:
The match can be partial.